lib/rules/wrong-quotes-for-template-string.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview Warn when using template string syntax in regular strings
+ * @author Jeroen Engels
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Warn when using template string syntax in regular strings",
+            category: "Possible Errors",
+            recommended: true
+        },
+
+        schema: []
+    },
+
+    create: function(context) {
+        var regex = /\$\{.*\}/;
+
+        return {
+            Literal: function(node) {
+                if (typeof node.value === "string" && regex.test(node.value)) {
+                    context.report(node, "Possible use of template string inside a regular string.");
```

Comment:
Ha, that's what I usually use, but I checked an existing rule to be consistent with the project style, and must have copied it from one that wasn't updated :sweat_smile:

Am I allowed to use object literal syntax?

```js
context.report({
    node,
    message: "Possible use of template string inside a regular string."
});
```

Sidenote: for my own plugins, I found it pretty nice to have the following syntax, but didn't use it as it isn't the norm here.
```js
const create = (context) => {} // of function create() {}, whatever

const schema = [];

module.exports = {
    create,
    meta: {
        schema,
        docs: {
            description: "Warn when using template string syntax in regular strings",
            category: "Possible Errors",
            recommended: true
        }
    }
};
```

Don't plan on adding it here now, but food for thought.